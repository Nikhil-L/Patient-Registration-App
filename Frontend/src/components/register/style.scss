.base-container {
    padding-top: 30px;
    width: 80%;
    float: right;
    display: flex;
    flex-direction: column;
    align-items: center;
    .header {
      font-size: 24px;
      font-family: "Open Sans", sans-serif;
    }
  
    .content {
      display: flex;
      flex-direction: column;
  
      .image {
        width: 21em;
        img {
          width: 100%;
          height: 100%;
        }
      }
  
      .form {
        margin-top: 5px;
        display: flex;
        flex-direction: column;
        align-items: center;
        margin-right: 100px;
  
        .form-group {
          display: flex;
          flex-direction: column;
          align-items: flex-start;
          width: fit-content;
          .text-error{
            color: red;
          }
          select{
              margin-left: 70px;
              margin-right: 110px;
          }
          label {
            font-size: 20px;
          }
          input {
            margin-top: 6px;
            min-width: 18em;
            height: 37px;
            padding: 0px 10px;
            font-size: 16px;
            font-family: "Open Sans", sans-serif;
            background-color: #f3f3f3;
            border: 0;
            border-radius: 4px;
            margin-bottom: 25px;
            transition: all 250ms ease-in-out;
            &:hover {
              background-color: #ffffff;
              //box-shadow: 0px 0px 14px 0.3px #0e81ce96;
            }
  
            &:focus {
              outline: none;
              box-shadow: 0px 0px 12px 0.8px #3474dbb2;
            }
          }
        }
      }
    }
    .button{
      color : white;
      background-color: black;
      width : 100px;
      height: 35px;
      align-items: center;
      align-self: center;
      margin-left: 90px;
      margin-top: 5px;
      margin-bottom: 10px;
    }
  
    
  }