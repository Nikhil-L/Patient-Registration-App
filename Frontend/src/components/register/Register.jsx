import axios from "axios";
import React from "react";
import { DateOfBirth }  from "../dateOfBirth/DateOfBirth";
import { RegistrationDate }from "../registrationDate/RegistrationDate";
import "./style.scss";

export class Register extends React.Component{

  /**
    * Initializes the props
    * @param {*} props 
    */
  constructor(props) {
      super(props);
  
      this.state = {
        input: {"firstName":"", "lastName":"", "gender":"Male"},
        errors: {"firstName":""}
      };

      this.handleChange = this.handleChange.bind(this);
      this.handleSubmit = this.handleSubmit.bind(this);
      this.convert = this.convert.bind(this)
  }

  /**
   * This is to handle the changes made on the webbrowser
   * @param {*} event 
   * @param {*} name the event name that triggered 
   * @param {*} istriggered checks whether it was triggered automatically or foced triggered
   */
    handleChange(event, name,  istriggered = true) {
        let input = this.state.input;
        istriggered?input[event.target.name] = event.target.value:input[name] = event;
        this.setState({input:input});
      }

    /**
     * To convert the javascript date into sql date
     * @param {*} dt date
     * @returns 
     */
    convert(dt){
      var date = new Date(dt);
      var month=("0"+(date.getMonth()+1)).slice(-2);
      var day = ("0"+date.getDate()).slice(-2);
      return [date.getFullYear(), month, day].join("-")
    }
  
    /**
     * This method is used to handle the submissions
     * @param {*} event 
     */
    handleSubmit(event) {
      event.preventDefault();
    
      if(this.validate()){

          let input = this.state.input;
          input["registrationDate"] = this.convert(input["registrationDate"]);
          input["dateOfBirth"] = this.convert(input["dateOfBirth"]);
          this.setState({input:input});
          let url = "http://localhost:8082/patient_registration_war/webapi/register";
    
          axios.post(url, input).then((response)=>{window.location.reload()}).catch(error => {
              if(!error.response)
                window.location = "/dberror"
            });
          
      }
    }
  
    /**
     * This is to validate the registration form
     * @returns boolean value
     */
    validate(){
      let input = this.state.input;
      let errors = {};
      let isValid = true;
      
      if (!input["firstName"]) {
        isValid = false;
        errors["firstName"] = "Please enter your First Name.";
      }

      if (!input["lastName"]) {
          isValid = false;
          errors["lastName"] = "Please enter your Last Name.";
      }
    
  
      this.setState({
        errors: errors
      });
  
      return isValid;
    }
  
  /**
  * This is used to render the component to the webpage
  * @returns a html code 
  */
  render=()=>{
      return(
          <div className="base-container" ref={this.props.containerRef}>
              <form onSubmit={this.handleSubmit}>
                  <div className="content">
                      <div className="form">
                          <div className="form-group">
                            <div className="text-error">{this.state.errors.firstName}</div>
                              <label htmlFor="lastName">First Name</label>
                              <input 
                              type="text"
                              name="firstName"
                              value={this.state.input.firstname}
                              onChange={this.handleChange}
                              placeholder="First Name" />
                          </div>
                          <div className="form-group">
                            <div className="text-error">{this.state.errors.lastName}</div>
                              <label htmlFor="lastName">Last Name</label>
                              <input
                              type="text"
                              name="lastName"
                              value={this.state.input.lastname}
                              onChange={this.handleChange}
                              placeholder="Last Name" />
                          </div>
                          <div className="form-group">
                            <div>
                              <label className = "gender" htmlFor="gender">Gender</label>
                              <select
                                className="gender" 
                                name="gender" 
                                value={this.state.input.gender}  
                                onChange={this.handleChange}>
                                  <option value="Male">Male</option>
                                  <option value="Female">Female</option>
                                  <option value="Others">Others</option>
                              </select>
                            </div>
                            <br></br>
                          </div>
                          <RegistrationDate 
                          name="registrationDate" 
                          value={this.state.input.RegistrationDate} 
                          handleChange={(date)=>this.handleChange(date, "registrationDate", false)}/>
                          <DateOfBirth
                          name="dateOfBirth"
                          value = {this.state.input.DateOfBirth} 
                          handleChange={(date)=> this.handleChange(date, "dateOfBirth", false)}/>
                      </div>
                  </div>
                  <button type="submit" value="submit" className="button">
                  Register
                  </button>
              </form>
          </div>
      );
  }
  
}