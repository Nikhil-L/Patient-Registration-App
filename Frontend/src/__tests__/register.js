import React from "react";
import {Register} from "../components/register/Register"
import renderer from "react-test-renderer"
import {shallow} from "enzyme"

/**
 * This is to test the Register component
 */
describe("Register test", ()=>{

  /**
   * Snapshot testing
   */
  it("matches a snapshot", ()=>{
    const tree = renderer.create(<Register/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  /**
   * register event testing
   */
  it('event testing', ()=>{
    let wrapper = shallow(<Register/>);
    wrapper.find('form').props().onSubmit({preventDefault: jest.fn()});
    wrapper.find({name:'firstName'}).simulate('change', {target:
      {name:'firstName', value:'Arun'}})
    wrapper.find({name:'lastName'}).simulate('change', {target:
      {name:'lastName', value:'Kumar'}})
    wrapper.find({name:'gender'}).simulate('change', {target:
      {name:'gender', value:'Male'}})
    wrapper.find("RegistrationDate").props().handleChange('03/07/2000');
    wrapper.find("DateOfBirth").props().handleChange('03/07/2000');
    wrapper.find('form').props().onSubmit({preventDefault: jest.fn()});
  })
})