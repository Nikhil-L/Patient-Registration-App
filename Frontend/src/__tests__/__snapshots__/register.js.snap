// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register test matches a snapshot 1`] = `
<div
  className="base-container"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="form"
      >
        <div
          className="form-group"
        >
          <div
            className="text-error"
          >
            
          </div>
          <label
            htmlFor="lastName"
          >
            First Name
          </label>
          <input
            name="firstName"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <div
            className="text-error"
          />
          <label
            htmlFor="lastName"
          >
            Last Name
          </label>
          <input
            name="lastName"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <div>
            <label
              className="gender"
              htmlFor="gender"
            >
              Gender
            </label>
            <select
              className="gender"
              name="gender"
              onChange={[Function]}
              value="Male"
            >
              <option
                value="Male"
              >
                Male
              </option>
              <option
                value="Female"
              >
                Female
              </option>
              <option
                value="Others"
              >
                Others
              </option>
            </select>
          </div>
          <br />
        </div>
        <div
          className="form-group"
        >
          <div>
            <label
              name="Registration Date"
            >
              Registration Date
            </label>
          </div>
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className=""
                disabled={false}
                name="startDate"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                type="text"
                value="07/07/2021"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div>
            <label
              name="DateOfBirth"
            >
              Date of Birth
            </label>
          </div>
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className=""
                disabled={false}
                name="startDate"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                type="text"
                value="07/07/2021"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="button"
      type="submit"
      value="submit"
    >
      Register
    </button>
  </form>
</div>
`;
